### PR TITLE
Fix Blazar unit tests

### PR DESCRIPTION
**Pull request by AnishReddyRavula**
Fix blazar unit tests by cherry-picking few commits from upstream
most of the methods expect project-id key in the lease dict

### CODE REVIEW

**Review by super-cooper — APPROVED** [review]
LGTM, but I would wait for Mark's review before merging

**Review by Mark-Powers — APPROVED** [review]
Looks great!

**Pull request merged**
